### PR TITLE
Fix wrong emoji urls.

### PR DESCRIPTION
**Pull request by agarant**
Some emojis like: ❤️,✈️ and 🛰️have a length > 1 but their url should only have one part. Because of it, some urls were incorrectly built and therefore not rendering in the final pdf document.

This PR:
- Add tests for fetch emojis
- Fix this case

### CODE REVIEW

File: src/utils/emoji.js [hunk 1]
```diff
@@ -10,10 +10,11 @@ const regex = emojiRegex();
 
 const reflect = promise => (...args) => promise(...args).then(v => v, e => e);
 
-const fetchEmojiImage = reflect(resolveImage);
+const makeFetchEmojiImage = () => reflect(resolveImage);
```
Comment (diegomura) [inline src/utils/emoji.js]: Can you explain the reason of this change?

File: src/utils/emoji.js [hunk 2]
```diff
@@ -10,10 +10,11 @@ const regex = emojiRegex();
...
 
 const getCodePoints = string =>
   Array.from(string)
+    .filter(x => x !== '️')
```
Comment (diegomura) [inline src/utils/emoji.js]: And this 😅

**Comment by diegomura** [comment]
Thanks!! I wasn’t aware of this issue. I appreciate the PR. I made some questions but everything looks good for me

File: src/utils/emoji.js (hunk 2 above)
Comment (agarant) [reply src/utils/emoji.js]: Yes, this is not really obvious, I will add a comment. This is the fix. The picture below should make it clearer.

image

File: src/utils/emoji.js (hunk 1 above)
Comment (agarant) [reply src/utils/emoji.js]: When it is a constant, it is not possible to mock resolveImage as it uses the real version before jest is able to mock it. I will add a comment.

**Comment by diegomura** [comment]
Thanks! Will appreciate those comments. But overall it looks great!

Just so you know, you can now mock `fetch` for testing cases such as these (as [here](https://github.com/diegomura/react-pdf/blob/master/tests/image.test.js#L76)). Your approach is good for me though!

I'm not sure if the last headers changes will affect how emojis are being loaded, but I'll take a look after this get's merged and fix it in case it's needed

**Comment by agarant** [comment]
@diegomura Good to know! Thank you for this library and being so quick at responding to PRs! 👍

**Comment by diegomura** [comment]
Thank you for taking care of this! All contributions are very welcomed 😄

**Pull request merged**